### ISSUE TITLE
Priority doesn't seem to behave as expected

### ISSUE THREAD

**Issue by George3d6** [Bogdanp/dramatiq#508]
# Issues

Note -- I'm running Django-dramatiq

I need to use dramatiq to execute higher-priority tasks before lower-priority tasks, and I have a fairly long queue of tasks (maybe a few thousand running on only a 6 to 8 workers). However, I have very few high-priority tasks, and I'd expect those to **always** be executed first. But, the behavior I observe is that dramatiq will keep choosing to run enqueued actors with lower priorities (sometimes).

As far as I understood from reading a bunch of issues what I'd actually need here is "queue priority" or some such, which is only supported by RabbitMQ. But I'd be nice to double-check and see I'm not doing something wrong, and also, if priority is indeed limited to a specific situation (e.g. priority of resource allocation to already running tasks) I'd be nice if the documentation was more specific about this.

## Checklist

* [x] Does your title concisely summarize the problem?
* [ ] Did you include a minimal, reproducible example?
* [x] What OS are you using?
* [ ] What version of Dramatiq are you using?
* [ ] What did you do?
* [ ] What did you expect would happen?
* [ ] What happened?

## What OS are you using?

Ubuntu 22.04 LTS

## What version of Dramatiq are you using?

1.13.0

## What did you do?

I need to use dramatiq to execute higher-priority tasks before lower-priority tasks, and I have a fairly long queue of tasks (maybe a few thousand running on only a 6 to 8 workers). However, I only enqueued a few high-priority tasks

## What did you expect would happen?

I have very few high-priority tasks, and I'd expect those to **always** be executed first.

## What happened?

The behavior I observe is that dramatiq will keep choosing to run enqueued actors with lower priorities (sometimes).

**Comment by LincolnPuzey** [comment]
@George3d6 Yes you are right, "queue priority" is only supported by RabbitMQ, and the `priority` option only effects messages that have been taken off the queue, and are waiting in-memory for a worker thread.
